### PR TITLE
remove deprecated marching_cubes '_lorensen' option

### PR DESCRIPTION
**Pull request by grlee77**
## Description

The legacy marching cubes code was deprecated in v0.17.0 and according to the warning, it was scheduled for removal in v0.19.0. That didn't happen, so let's remove it for v0.20.0 instead.

Note that `mesh_surface_area` was not modified. It was just moved out of `_marching_cubes_classic.py` so that file could be removed.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Review by rfezzani — APPROVED** [review]
Thank you @grlee77 for spotting this, we in fact forgot about it :sweat_smile:! Shouldn't it be backported too?

**Comment by mkcor** [comment]
> Shouldn't it be backported too?

@rfezzani not particularly, since @grlee77 wrote "let's remove it for v0.20.0 instead" :slightly_smiling_face:

File: skimage/measure/_marching_cubes_lewiner.py [hunk 1]
```diff
@@ -43,11 +42,8 @@ def marching_cubes(volume, level=None, *, spacing=(1., 1., 1.),
         Whether to allow degenerate (i.e. zero-area) triangles in the
         end-result. Default True. If False, degenerate triangles are
         removed, at the cost of making the algorithm slower.
-    method: str, optional
-        One of 'lewiner', 'lorensen' or '_lorensen'. Specify which of
-        Lewiner et al. or Lorensen et al. method will be used. The
-        '_lorensen' flag correspond to an old implementation that will
-        be deprecated in version 0.19.
+    method: {'lewiner', 'lorensen'}, optional
+        Whether the method of Lewiner et al. or Lorensen et al. will be  used.
```
Comment (mkcor) [inline skimage/measure/_marching_cubes_lewiner.py:46]:
```suggestion
        Whether the method of Lewiner et al. or Lorensen et al. will be used.
```

File: skimage/measure/_marching_cubes_lewiner.py [hunk 2]
```diff
@@ -272,3 +255,45 @@ def _get_mc_luts():
...
+    """
+    Compute surface area, given vertices & triangular faces
+
+    Parameters
+    ----------
+    verts : (V, 3) array of floats
+        Array containing (x, y, z) coordinates for V unique mesh vertices.
+    faces : (F, 3) array of ints
+        List of length-3 lists of integers, referencing vertex coordinates as
+        provided in `verts`
```
Comment (mkcor) [inline skimage/measure/_marching_cubes_lewiner.py:270]:
```suggestion
        provided in `verts`.
```

File: skimage/measure/_marching_cubes_lewiner.py [hunk 3]
```diff
@@ -272,3 +255,45 @@ def _get_mc_luts():
                 )
 
     return mcluts.THE_LUTS
+
+
+def mesh_surface_area(verts, faces):
+    """
+    Compute surface area, given vertices & triangular faces
```
Comment (mkcor) [inline skimage/measure/_marching_cubes_lewiner.py:262]:
```suggestion
    """Compute surface area, given vertices and triangular faces.
```

File: skimage/measure/_marching_cubes_lewiner.py [hunk 4]
```diff
@@ -272,3 +255,45 @@ def _get_mc_luts():
...
+    `skimage.measure.marching_cubes`. For unit correct output, ensure correct
+    `spacing` was passed to `skimage.measure.marching_cubes`.
+
+    This algorithm works properly only if the ``faces`` provided are all
+    triangles.
+
+    See Also
+    --------
+    skimage.measure.marching_cubes
+    skimage.measure.marching_cubes_classic
```
Comment (mkcor) [inline skimage/measure/_marching_cubes_lewiner.py:289]:
```suggestion
```

**Comment by rfezzani** [comment]
> > Shouldn't it be backported too?
> 
> @rfezzani not particularly, since @grlee77 wrote "let's remove it for v0.20.0 instead" slightly_smiling_face

^^ I was just wondering why we can't simply backport it instead of waiting for v0.20/v1.0

**Comment by grlee77** [comment]
> ^^ I was just wondering why we can't simply backport it instead of waiting for v0.20/v1.0

I am not strongly against it, but it didn't seem like it provides much concrete benefit as a backport and it would technically be an API change on a point release.

**Review by mkcor — APPROVED** [review]

**Pull request merged**